clippy_lints/src/booleans.rs

```diff
@@ -346,11 +346,18 @@ fn simplify_not(cx: &LateContext<'_>, expr: &Expr<'_>) -> Option<String> {
                 _ => None,
             }
             .and_then(|op| {
-                Some(format!(
-                    "{}{op}{}",
-                    snippet_opt(cx, lhs.span)?,
-                    snippet_opt(cx, rhs.span)?
-                ))
+                let lhs_snippet = snippet_opt(cx, lhs.span)?;
+                let rhs_snippet = snippet_opt(cx, rhs.span)?;
+
+                if let (ExprKind::Cast(..), BinOpKind::Ge) = (&lhs.kind, binop.node) {
+                    if !(lhs_snippet.starts_with('(') && lhs_snippet.ends_with(')')) {
```

Comment:
Could you swap the order of these two so that the rare condition is processed earlier? (Performance meow meow)

Also, does this need to check for casts and >=? Are these the only variables where the suggestion is incorrect?

Reply:
> Could you swap the order of these two so that the rare condition is processed earlier? (Performance meow meow)

done

> Also, does this need to check for casts and >=? Are these the only variables where the suggestion is incorrect?

no, `>=` is parsed as a single token, and is never seen as the start of a sequence of generic arguments. I believe this is the only operation where the naive transformation produces a result that does not parse.